openupgrade_scripts/scripts/hr/16.0.1.1/post-migration.py

```diff
@@ -182,21 +103,47 @@ def fill_master_department_id(cr):
     )
 
 
-@openupgrade.migrate()
-def migrate(env, version):
-    warn_about_dataloss(
+def _hr_plan_activity_type_m2m_to_o2m(env):
+    """Before, the activities in plans (hr.plan) were linked to the
+    plan_activity_type_ids field with an m2m field, now the field is the same but it
+    is an o2m. We define the data in the table hr_plan_activity_type according to the
+    table ou_legacy_16_0_hr_plan_hr_plan_activity_type_rel that we have defined
+    previously in pre-migration to ensure that no data is lost.
```

Comment:
where? in which pre-migration? in which module?

Reply:
All of this is done in `hr` pre-migration, specifically in https://github.com/OCA/OpenUpgrade/blob/0df0cf282646b82ac4b9cec9079d16e9cca9e13c/openupgrade_scripts/scripts/hr/16.0.1.1/pre-migration.py#L55